### ISSUE TITLE
Upgrade RSpec

### ISSUE THREAD

**Issue by codenamev** [reenhanced/gitreflow#162]
As @pboling [has pointed out](https://github.com/reenhanced/gitreflow/pull/161#issuecomment-209031976), we can use the [transpec](https://github.com/yujinakayama/transpec) gem to do a lot of automated translations to the new `allow` syntax.

**Comment by simonzhu24** [comment]
Ready for a PR =)

**Comment by simonzhu24** [comment]
@codenamev I think we can close this issue now.

**Comment by codenamev** [comment]
I believe we still need to upgrade RSpec itself? @pboling should have details on the issues he faced.  He locked us in at 3.0 in a previous PR, so I think we still need to do a `bundle update rspec` here and do some sanity checks on the specs.

**Comment by codenamev** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;nzFhAUqt&#x2F;13-upgrade-rspec"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Upgrade RSpec</a>

**Comment by pboling** [comment]
Actually I didn't lock to 3.0, I actually unlocked so we can upgrade.

Old version constraint was `~> 3.0.0`, and I changed to `~> 3.0` meaning that with a simple `bundle update rspec` it will now install the latest 3.4 version, and put that in the Gemfile.lock.  @simonzhu24 will you do that, and see if all the tests pass?  There were so many failures before the syntax fix that I didn't bother to look at the causes.
